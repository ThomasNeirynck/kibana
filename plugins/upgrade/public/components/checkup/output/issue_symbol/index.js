import classNames from 'classnames';
import React from 'react';

import { DEPRECATION_ISSUE_LEVELS } from '../../../../lib/constants';


export function IssueSymbol({ level }) {
  const levelToClassNameMap = {
    none: 'kuiIcon--success fa-check',
    info: 'kuiIcon--info fa-info',
    warning: 'kuiIcon--warning fa-bolt',
    critical: 'kuiIcon--error fa-warning',
  };

  const iconClasses = classNames('kuiIcon', levelToClassNameMap[level]);

  return (
    <span className='kuiEventSymbol'>
      <span className={ iconClasses } />
    </span>
  );
}

IssueSymbol.propTypes = {
  level: React.PropTypes.oneOf(DEPRECATION_ISSUE_LEVELS),
};

IssueSymbol.defaultProps = {
  level: null,
};
