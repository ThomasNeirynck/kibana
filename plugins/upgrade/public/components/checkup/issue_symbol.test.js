import { shallow } from 'enzyme';
import React from 'react';

import { IssueSymbol } from './issue_symbol';

jest.mock('ui/chrome', () => {}, { virtual: true });
jest.mock('ui/notify/notifier', () => ({
  Notifier: function () {},
}), { virtual: true });


describe('IssueSymbol', () => {
  test('renders a symbol for the "critical" level', () => {
    const component = (
      <IssueSymbol level="critical" />
    );

    expect(shallow(component)).toMatchSnapshot();
  });

  test('renders a symbol for the "info" level', () => {
    const component = (
      <IssueSymbol level="info" />
    );

    expect(shallow(component)).toMatchSnapshot();
  });

  test('renders a symbol for the "none" level', () => {
    const component = (
      <IssueSymbol level="none" />
    );

    expect(shallow(component)).toMatchSnapshot();
  });

  test('renders a symbol for the "warning" level', () => {
    const component = (
      <IssueSymbol level="warning" />
    );

    expect(shallow(component)).toMatchSnapshot();
  });
});
