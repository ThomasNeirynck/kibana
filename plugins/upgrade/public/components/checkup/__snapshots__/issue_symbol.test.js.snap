// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IssueSymbol renders a symbol for the "critical" level 1`] = `
<span
  className="kuiEventSymbol"
>
  <span
    className="kuiIcon kuiIcon--error fa-warning"
  />
</span>
`;

exports[`IssueSymbol renders a symbol for the "info" level 1`] = `
<span
  className="kuiEventSymbol"
>
  <span
    className="kuiIcon kuiIcon--info fa-info"
  />
</span>
`;

exports[`IssueSymbol renders a symbol for the "none" level 1`] = `
<span
  className="kuiEventSymbol"
>
  <span
    className="kuiIcon kuiIcon--success fa-check"
  />
</span>
`;

exports[`IssueSymbol renders a symbol for the "warning" level 1`] = `
<span
  className="kuiEventSymbol"
>
  <span
    className="kuiIcon kuiIcon--warning fa-bolt"
  />
</span>
`;
